spec/policy/conditional/operation_spec.lua

```diff
@@ -0,0 +1,45 @@
+local Operation = require('apicast.policy.conditional.operation')
+
+describe('Operation', function()
+  describe('.new', function()
+    it('raises error with an unsupported operation', function()
+      local res, err = pcall(Operation.new, '1', 'plain', '<>', '1', 'plain')
+
+      assert.is_falsy(res)
+      assert.is_truthy(err)
+    end)
+  end)
+
+  describe('.evaluate', function()
+    it('evaluates ==', function()
+      assert.is_true(Operation.new('1', 'plain', '==', '1', 'plain'):evaluate({}))
+      assert.is_false(Operation.new('1', 'plain', '==', '2', 'plain'):evaluate({}))
+    end)
+
+    it('evaluates !=', function()
+      assert.is_true(Operation.new('1', 'plain', '!=', '2', 'plain'):evaluate({}))
+      assert.is_false(Operation.new('1', 'plain', '!=', '1', 'plain'):evaluate({}))
+    end)
+
+    it('evaluates values as plain text by default', function()
+      assert.is_true(Operation.new('1', nil, '==', '1', nil):evaluate({}))
+      assert.is_false(Operation.new('1', nil, '==', '2', nil):evaluate({}))
+    end)
+
+    it('evaluates liquid when indicated in the types', function()
+      local context = { var_1 = '1', var_2 = '2' }
+
+      local res_true = Operation.new(
+        '{{ var_1 }}', 'liquid', '==', '1', 'plain'
```

Comment:
Are we going to handle types somehow? What if the variable comes from nginx and is actually a number ?

Reply:
In that example, even if `var_1` was a number, the comparison would end up being between 2 strings, because `LiquidTemplateString:render()` returns a string.

I included a test that checks this. Also, I added `tostring()` in the `==` and `!=` operations in `Operation` to make this explicit.